### PR TITLE
Implement userstore health check

### PR DESCRIPTION
**Pull request by ivantha**
## Purpose
$subject.

## Approach
Added a new parameter to the response of the health check stating if a particular userstore is healthy.

## References
https://github.com/wso2/product-is/issues/9344

### CODE REVIEW

File: components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java [hunk 1]
```diff
@@ -121,24 +124,27 @@ public Properties checkHealth() throws BadHealthException {
 
             } while (secondaryRealmConfiguration != null);
         }
-        return userstores.toArray(new String[userstores.size()]);
+        return userstores.toArray(new String[0]);
     }
 
-    protected void validateUSM(UserStoreManager userStoreManager, String domainName, List<HealthCheckError> errors) {
+    protected void validateUSM(UserStoreManager userStoreManager, String domainName, Properties properties,
```
Comment (somindatommy) [inline components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java:130]: why do we need to change the protected method signature?

Comment (somindatommy) [inline components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java:130]: why do we need to change the protected method signature?

File: components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java [hunk 2]
```diff
@@ -136,8 +139,28 @@ protected void validateUSM(UserStoreManager userStoreManager, String domainName,
...
             }
-            errors.add(new HealthCheckError(Constants.ErrorCodes.ERROR_USERSTORE_CONNECTIVITY_BY_IS_USER_EXISTING
-                    , "Error while checking health of USM with domain: " + domainName,
+            errors.add(new HealthCheckError(Constants.ErrorCodes.ERROR_USERSTORE_CONNECTIVITY_BY_IS_USER_EXISTING,
+                    "Error while checking health of USM with domain: " + domainName,
+                    Utils.getRootCauseMessage(e)));
+        }
+    }
+
+    protected void validateUSM(UserStoreManager userStoreManager, String domainName, Properties properties,
```
Comment (somindatommy) [inline components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java:148]: AFAIK the implementation of previous method is wrong right? If so, shall we deprecate the previous method? And add a method comment to the new method? WDYT?

Comment (ivantha) [reply components/org.wso2.carbon.healthcheck.api.core/src/main/java/org/wso2/carbon/healthcheck/api/core/impl/SuperTenantUSHealthChecker.java:148]: +1. I'll do that.

**Review by vihanga-liyanage — APPROVED** [review]

**Pull request merged**